src/runtime/includes/loaderQueue.ts

```diff
@@ -62,7 +62,7 @@ function loadLocaleQueue(locale: string, localeQueue: MessagesLoader[]) {
 
 const activeFlushes: { [key: string]: Promise<void> } = {};
 
-export function flush(locale: string): Promise<void> {
+export async function flush(locale: string): Promise<void> {
```

Comment:
This will change runtime behaviour, because an async function always wraps its returned value in a promise, even if it's a promise. So, comparing the return of one flush() with another, even if they return the same promise, will result in `false`. 

We should return `Promise.resolve()` on line 71 and remove the `async` modifier.

FWIW I think we only compare two `flush()` values in the tests.

Reply:
1.
Yes. I decieded to accpet your suggestion.
However, I want to note that `.then()` and `await ...` will unwrap all nested promises.
So it is perfectly identical `async () => await Promise.resolve()`, `async () => await undefined` and `() => Promise.resolve()`.
`async () => Promise.resolve()` is almost the same in semantics, except those above are consuming one tick in microtask queue, while it consumes two ticks. You can manage the style for [`return await ...` vs `return ...`](https://eslint.org/docs/rules/no-return-await) and [forcing to add `async` or not](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/promise-function-async.md) by eslint.